Riot/Modules/Room/RoomViewController.m

```diff
@@ -4342,49 +4363,48 @@ - (void)roomTitleView:(RoomTitleView*)titleView recognizeTapGesture:(UITapGestur
                         [self refreshRoomTitle];
                         [self refreshRoomInputToolbar];
                     }
+                    break;
                 }
-                
-            }];
-        }
-        else
-        {
-            [self joinRoom:^(BOOL succeed) {
-                
-                if (succeed)
-                {
-                    [self refreshRoomTitle];
-                }
-                
-            }];
-        }
+                case MXKRoomViewControllerJoinRoomResultFailureRoomEmpty:
+                    [self declineRoomInvitation];
```

Comment:
In case of any error maybe we should show an alert telling the user what happened? Could pretty pretty frustrating to not know why a room join has failed. Also in the case of auto declining the room I think the user would have a better understanding of why the invitation disappeared, having gotten a message first.

Reply:
I 100% agree, the error part is (already was) handled in the [kit superclass here](https://github.com/matrix-org/matrix-ios-kit/blob/48ae9cc0a828197c155e525943e5b30378b5a700/MatrixKit/Controllers/MXKRoomViewController.m#L979). Not great not terrible 😁